### DISCUSSION TITLE
v5.0.0-alpha.0

### DISCUSSION THREAD

**Discussion by keithamus** [chaijs/chai#1500] [Announcements]
This release is the first alpha release for Chai 5. The changes from v4.x:

 - This package now uses ESM modules. CommonJS environments are no longer supported. If you wish to use a CommonJS environment, please continue to use Chai 4.
 - Drop support for NodeJS versions less than 16. If you are using NodeJS v4-v15, please continue to use Chai 4.
 - Drop support for Internet Explorer 10 & 11. If you wish to test against Internet Explorer 10 or 11 please continue to use Chai 4.
 - Drop support for Safari versions less than 14. If you wish to test against Safari v10-v13  please continue to use Chai 4.
 - Drop support for EdgeHTML. If you wish to test against Edges older than 18 (as in, not Chromium Edge)  please continue to use Chai 4.
 - Drop support for Firefox versions less than 102. If you wish to test against Firefox 60 through to 101  please continue to use Chai 4.
 - Drop support for Chrome versions less than 108. If you wish to test against Chrome 68 through 107  please continue to use Chai 4.

<hr /><em>This discussion was created from the release <a href='https://github.com/chaijs/chai/releases/tag/v5.0.0-alpha.0'>v5.0.0-alpha.0</a>.</em>

**Comment by koddsson** [comment]
I made a [draft PR](https://github.com/modernweb-dev/web/pull/2176) to `modernweb-dev/web` with `chai@next` and it's working great! I'm happy to drop `@esm-bundle/chai` in favour of `chai` 🎉 

I figured I'd keep the draft PR until the version is out of prerelease 👍🏻

**Reply by Westbrook** [reply]
@keithamus any thoughts on when we might see a full release here? Would be great to be able to adopt these updates in the `@web/*` family of products!

**Reply by keithamus** [reply]
I’d like to get the dependencies migrated to ESM and cut new chai releases with each one. So it’ll be a while yet. If you would like to help, PRs are very much welcome!

**Reply by Westbrook** [reply]
Not to be taken as _volunteering_, but do you have a list of what dependencies you mean? If nothing else to understand the scale of work between here and there.

**Reply by keithamus** [reply]
https://github.com/chaijs/chai/blob/4.x.x/package.json#L43-L49 It would be these ones. 

I’d like to update these all to modernise them to ESM/typescript as I think it would help us make changes more easily in chai core, and I’d like to do it without too many breaking changes. 

`get-func-name` and `type-detect` can probably be dropped as dependencies because they are mostly to correct backwards compatibility issues. `deep-eql` is our most up to date dep, I believe @koddsson did some work on turning it into esm/TS. I have a working branch for `loupe`. This leaves `pathval`, `assertion-error` and `check-error`. I believe each one is <100LOC so I don’t think it’s going to be a load of work to convert them.

**Reply by koddsson** [reply]
> `deep-eql` is our most up to date dep, I believe @koddsson did some work on turning it into esm/TS.

Oohh I can continue that work. I see there's an old PR open. I'll try to tackle the rest as well once `deep-eql` is done.

**Reply by koddsson** [reply]
Yo yo yo; I've created stacked PRs that 1. Remove Karma in favour of Web Test Runner (https://github.com/chaijs/deep-eql/pull/94), 2. Convert the project to an ES Module (https://github.com/koddsson/deep-eql/pull/1) and then a bit of 3 refactoring (https://github.com/koddsson/deep-eql/pull/2)

Let me know what you think. I made a bunch of assumptions which may be wrong, so just comment on the PRs and I'll make those changes.

**Reply by koddsson** [reply]
Looks like `assertion-error` is already ESM and TypeScript.

https://github.com/chaijs/assertion-error/blob/08a1f16ece0c5d4b916c9ce1479a776df0e203e0/package.json#L21

**Reply by koddsson** [reply]
Tracking it here (I'll update the comment as we make progress on this).

| Project         | ESM | TypeScript | Updated in chai@v5 branch | 
|-----------------|-----|------------|-------------------------|
| deep-eql        | ✅   | ❌          | ✅
| check-error     | ✅   | ❌          | ✅
| assertion-error | ✅   | ✅          | ❌
| pathval         | ✅   | ❌          | ✅
| loupe         | ✅   | ✅          | ✅

**Reply by Westbrook** [reply]
Any progress on this?

**Reply by koddsson** [reply]
Yup! I've been working on this lately. `deep-eql` is updated in the chai@v5 branch, there's a PR open for `assertion-error` and we're finalising the `loupe` migration to ESM.

**Comment by voxpelli** [comment]
> CommonJS environments are no longer supported. If you wish to use a CommonJS environment, please continue to use Chai 4

CJS is still supported in Chai 5 if imported through `const chai = await import('chai')`, right?

**Reply by keithamus** [reply]
Hopefully so. Our main goal will be to ensure that `import chai from 'chai'` will work.
